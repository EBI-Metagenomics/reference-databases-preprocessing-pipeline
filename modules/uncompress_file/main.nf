
process UNCOMPRESS_FILE {

    label 'light'

    input:
    path cmp_file
    val uncmp_name

    output:
    path("${uncmp_name}"), emit: uncmp_file

    """
    gunzip -c -f $cmp_file > $uncmp_name
    """
}