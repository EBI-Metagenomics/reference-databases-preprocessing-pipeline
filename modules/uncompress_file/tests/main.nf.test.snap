{
    "uncmp_file": {
        "content": [
            [
                "test.fasta:md5,68c60c9c9b78c0244f05e78fac9926a6"
            ]
        ],
        "timestamp": "2023-12-12T14:11:23.570462108"
    },
    "remote_uncmp_file": {
        "content": [
            [
                "test.fasta:md5,dafd38f5454b54fbea38245d773062a5"
            ]
        ],
        "timestamp": "2023-12-12T14:18:14.059557772"
    }
}